cocos/2d/CCNode.cpp

```diff
@@ -1236,29 +1191,32 @@ const kmMat4& Node::getNodeToParentTransform() const
 
         // Build Transform Matrix
         // Adjusted transform calculation for rotational skew
-        kmScalar mat[] = { cy * _scaleX, sy * _scaleX,     0,  0,
-                        -sx * _scaleY, cx * _scaleY,    0,  0,
+        _transform = { cy * _scaleX, sy * _scaleX, 0,  0,
+                      -sx * _scaleY, cx * _scaleY, 0,  0,
                         0,  0,  1,  0,
                         x,  y,  0,  1 };
 
-        kmMat4Fill(&_transform, mat);
         // XXX: Try to inline skew
         // If skew is needed, apply skew and then anchor point
         if (needsSkewMatrix)
         {
-            kmMat4 skewMatrix = {    1, tanf(CC_DEGREES_TO_RADIANS(_skewY)), 0, 0,
-                                    tanf(CC_DEGREES_TO_RADIANS(_skewX)),1, 0, 0,
-                                    0,  0, 1, 0,
-                                    0,  0,  0, 1};
+            kmMat4 skewMatrix = { 1, tanf(CC_DEGREES_TO_RADIANS(_skewY)), 0, 0,
+                                  tanf(CC_DEGREES_TO_RADIANS(_skewX)), 1, 0, 0,
+                                  0,  0,  1, 0,
+                                  0,  0,  0, 1};
 
             kmMat4Multiply(&_transform, &skewMatrix, &_transform);
 
             // adjust anchor point
             if (!_anchorPointInPoints.equals(Point::ZERO))
             {
                 // XXX: Argh, kmMat needs a "translate" method
-                _transform.mat[12] += -_anchorPointInPoints.x;
-                _transform.mat[13] += -_anchorPointInPoints.y;
+                kmVec4 v = {-_anchorPointInPoints.x, -_anchorPointInPoints.y, 0, 1};
+                kmVec4Transform(&v, &v, &_transform);
+                _transform.mat[12] = v.x;
+                _transform.mat[13] = v.y;
+                _transform.mat[14] = v.z;
+                _transform.mat[15] = v.w;
```

Comment:
@ricardoquesada , I fixed it at https://github.com/cocos2d/cocos2d-x/pull/16736
Please review that PR. :)